File: promomap-web/src/app/models/Company.ts
import { Marker } from './Marker';

export class Company {
  private superId: string;
  private name: string;
  private description: string;
  private cnpj: string;
  private phone: string;
  private email: string;
  private location = new Marker();

  public setSuperId(superId: string): void {
    this.superId = superId;
  }

  public getSuperId(): string {
    return this.superId;
  }

  public setName(name: string): void {
    this.name = name;
  }

  public getName(): string {
    return this.name;
  }

  public setDescription(description: string): void {
    this.description = description;
  }

  public getDescription(): string {
    return this.description;
  }

  public setLocation(marker: Marker): void {
    this.location = marker;
  }

  public getLocation(): Marker {
    return this.location;
  }

  public setCnpj(cnpj: string): void {
    this.cnpj = cnpj;
  }

  public getCnpj(): string {
    return this.cnpj;
  }

  public setPhone(phone: string): void {
    this.phone = phone;
  }

  public getPhone(): string {
    return this.phone;
  }

  public setEmail(email: string): void {
    this.email = email;
  }

  public getEmail(): string {
    return this.email;
  }

  public equals(company: Company): boolean {
    if(company.getName() == this.getName()
      && company.getCnpj() == this.getCnpj()
      && company.getPhone() == this.getPhone()
      && company.getEmail() == this.getEmail()
      && company.getSuperId() == this.getSuperId()
      && company.getDescription() == this.getDescription()) {
      return true;
    } else {
      return false;
    }
  }
}

File: promomap/src/main/java/br/com/promomap/beans/persistence/Company.java
/*
 * @(#)Company.java 10 de mai de 2017 - 22:22:23
 *
 */
package br.com.promomap.beans.persistence;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import br.com.promomap.beans.transport.CompanyObject;
import br.com.promomap.beans.transport.UserObject;
import br.com.promomap.model.PersistenceBeanInterface;

/**
 * @author <a href="mailto:leandro.lucas_@hotmail.com">Leandro Lucas Santos</a>
 */
@Entity
@Table(name = "Company")
public class Company implements PersistenceBeanInterface<CompanyObject> {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column(name = "superId", nullable = false, unique = true)
	private String superId;

	@Column(name = "name")
	private String name;

	@Column(name = "description")
	private String description;

	@Column(name = "cnpj")
	private String cnpj;

	@Column(name = "phone")
	private String phone;

	@Column(name = "email")
	private String email;

	@OneToOne
	@JoinColumn(name = "location_id")
	private Location location;

	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "createdAt")
	private Date createdAt;

	@Column(name = "deleted", columnDefinition = "tinyint DEFAULT 0")
	private boolean deleted;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "deletedDate")
	private Date deletedDate;

	@Override
	public CompanyObject generateTransportObject() {
		CompanyObject companyO = new CompanyObject();
		companyO.setSuperId(getSuperId());
		companyO.setName(getName());
		companyO.setDescription(getDescription());
		companyO.setCnpj(getCnpj());
		companyO.setEmail(getEmail());
		companyO.setPhone(getPhone());
		companyO.setLocation(getLocation() == null ? null : getLocation().generateTransportObject());
		companyO.setUser(new UserObject().setSuperId(getUser().getSuperId()).setFirstName(getUser().getFirstName())
				.setLastName(getUser().getLastName()).setEmail(getUser().getEmail()));
		return companyO;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getSuperId() {
		return superId;
	}

	public void setSuperId(String superId) {
		this.superId = superId;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getCnpj() {
		return cnpj;
	}

	public void setCnpj(String cnpj) {
		this.cnpj = cnpj;
	}

	public String getPhone() {
		return phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public Location getLocation() {
		return location;
	}

	public void setLocation(Location location) {
		this.location = location;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Date getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(Date createdAt) {
		this.createdAt = createdAt;
	}

	public boolean isDeleted() {
		return deleted;
	}

	public void setDeleted(boolean deleted) {
		this.deleted = deleted;
	}

	public Date getDeletedDate() {
		return deletedDate;
	}

	public void setDeletedDate(Date deletedDate) {
		this.deletedDate = deletedDate;
	}

}
